### ISSUE TITLE
get_provider in configure_spilo.py fails on Docker Desktop Kubernetes

### ISSUE THREAD

**Issue by unclok** [zalando/spilo#673]
I'm trying to use spilo on my Docker Desktop Kubernetes, but I see below error all the time and configure_spilo.py stops with status code 1.

```
2021-12-12 03:47:50,865 - bootstrapping - INFO - Figuring out my environment (Google? AWS? Openstack? Local?)
2021-12-12 03:47:50,870 - bootstrapping - DEBUG - Starting new HTTP connection (1): 169.254.169.254
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 387, in _make_request
    six.raise_from(e, None)
  File "<string>", line 3, in raise_from
  File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 383, in _make_request
    httplib_response = conn.getresponse()
  File "/usr/lib/python3.6/http/client.py", line 1331, in getresponse
    response.begin()
  File "/usr/lib/python3.6/http/client.py", line 297, in begin
    version, status, reason = self._read_status()
  File "/usr/lib/python3.6/http/client.py", line 258, in _read_status
    line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
  File "/usr/lib/python3.6/socket.py", line 586, in readinto
    return self._sock.recv_into(b)
socket.timeout: timed out

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/requests/adapters.py", line 440, in send
    timeout=timeout
  File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 639, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "/usr/lib/python3/dist-packages/urllib3/util/retry.py", line 357, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "/usr/lib/python3/dist-packages/six.py", line 693, in reraise
    raise value
  File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 601, in urlopen
    chunked=chunked)
  File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 389, in _make_request
    self._raise_timeout(err=e, url=url, timeout_value=read_timeout)
  File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 309, in _raise_timeout
    raise ReadTimeoutError(self, url, "Read timed out. (read timeout=%s)" % timeout_value)
urllib3.exceptions.ReadTimeoutError: HTTPConnectionPool(host='169.254.169.254', port=80): Read timed out. (read timeout=2)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/scripts/configure_spilo.py", line 816, in <module>
    main()
  File "/scripts/configure_spilo.py", line 730, in main
    provider = os.environ.get('DEVELOP', '').lower() in ['1', 'true', 'on'] and PROVIDER_LOCAL or get_provider()
  File "/scripts/configure_spilo.py", line 270, in get_provider
    r = requests.get('http://169.254.169.254', timeout=2)
  File "/usr/lib/python3/dist-packages/requests/api.py", line 72, in get
    return request('get', url, params=params, **kwargs)
  File "/usr/lib/python3/dist-packages/requests/api.py", line 58, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/lib/python3/dist-packages/requests/sessions.py", line 520, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/lib/python3/dist-packages/requests/sessions.py", line 630, in send
    r = adapter.send(request, **kwargs)
  File "/usr/lib/python3/dist-packages/requests/adapters.py", line 521, in send
    raise ReadTimeout(e, request=request)
requests.exceptions.ReadTimeout: HTTPConnectionPool(host='169.254.169.254', port=80): Read timed out. (read timeout=2)
```

~~After I use `SPILO_PROVIDER=PROVIDER_LOCAL`, it finally started without error.~~

~~IMHO, it seems like when the read timeout got raised, request module raises an error and stops the code before the exception catch the error.~~

Updated: with DEVELOP=true, it is finally working. I couldn't find that information form https://github.com/zalando/spilo/blob/master/ENVIRONMENT.rst, it seems better to add about the relation between `DEVELOP` and `SPILO_PROVIDER=PROVIDER_LOCAL`

**Comment by CyberDem0n** [comment]
I am not sure which spilo version you are using, but pretty sure that something very old, because line numbers in configure_spilo.py are off by a lot.

**Comment by unclok** [comment]
https://github.com/zalando/spilo/blob/2.1-p3/kubernetes/spilo_kubernetes.yaml#L57
I used this script from the branch 2.1-p3. I think this branch is the newest branch.
The latest commit of this script is 14 May 2021, does this script need to be updated?

**Comment by CyberDem0n** [comment]
Well, usually people run spilo with [postgres-operator](https://github.com/zalando-incubator/postgres-operator).
This YAML manifest more or less exists for testing and/or for people who want to run it directly. In any case the manifest must be adjusted for your needs. It also applies to using the latest available image: https://github.com/zalando/spilo/releases

**Comment by unclok** [comment]
@CyberDem0n 
> In any case the manifest must be adjusted for your needs. It also applies to using the latest available image: https://github.com/zalando/spilo/releases
Oh, I got it. So this script is not fully managed and I needed to adjust by myself.
I understand now how it goes, but it would be easier to understand if there is any comment about it.
Also, I didn't think to use postgres-operator because I was just trying to make a small sandbox environment. But I feel like It could be easier to use spilo now.
Thanks for your explanation!
